src/components/home/Trend/types.ts

```diff
@@ -0,0 +1,13 @@
+// * 타입 정의
+export type TargetType = 'ALL' | 'FEMALE' | 'MALE' | 'TEEN'
+export type RankType = 'MANY_WISH' | 'MANY_RECEIVE' | 'MANY_WISH_RECEIVE'
+
+// * 유효한 TargetType인지 확인하는 함수
+export const isValidTargetType = (value: string): value is TargetType => {
+  return ['ALL', 'FEMALE', 'MALE', 'TEEN'].includes(value)
+}
+
+// * 유효한 RankType인지 확인하는 함수
+export const isValidRankType = (value: string): value is RankType => {
+  return ['MANY_WISH', 'MANY_RECEIVE', 'MANY_WISH_RECEIVE'].includes(value)
+}
```

Comment:
다만 좀더 제안을 드리자면...

타입 string 값들이 중복되는게 조금 아쉬워서요 ㅎㅎ

```suggestion
// * 타입 정의
export const TARGET_TYPE_LIST = ['ALL', 'FEMALE', 'MALE', 'TEEN'] as const
export const RANK_TYPE_LIST = ['MANY_WISH', 'MANY_RECEIVE', 'MANY_WISH_RECEIVE'] as const
export type TargetType = (typeof TARGET_TYPE_LIST)[number]
export type RankType = (typeof RANK_TYPE_LIST)[number]

// * 유효한 TargetType인지 확인하는 함수
export const isValidTargetType = (value: string): value is TargetType => {
  return TARGET_TYPE_LIST.includes(value as TargetType)
}

// * 유효한 RankType인지 확인하는 함수
export const isValidRankType = (value: string): value is RankType => {
  return RANK_TYPE_LIST.includes(value as RankType)
}
```

as const 키워드를 이용해 이렇게 작성해주시면 중복을 최소화할 수 있습니다 😁

Reply:
너무나 유익한 피드백 감사합니다!